### PR TITLE
Issue/4413 stripe onboarding screens

### PR DESCRIPTION
**Pull request by nbradbury**
This PR implements the UI layer for the following Stripe error states:

1. WCPayAccountPendingRequirementsState
2. WCPayAccountOverdueRequirementsState
3. WCPayAccountRejectedState
4. WCPayAccountUnderReviewState
5. WCPayInTestModeWithLiveAccountState

Note that I removed the button from the "pending requirements" state because that was the only state with a button, and it simply dismissed the fragment. Also, the screenshots below use the wrong title case and are missing the toolbar divider, but those have been added since taking the screenshots.

stripe

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt [hunk 1]
```diff
@@ -61,28 +63,55 @@ class CardReaderOnboardingFragment : BaseFragment(R.layout.fragment_card_reader_
...
+            is CardReaderOnboardingViewModel.OnboardingViewState.WCPayError ->
+                showWCPayErrorState(layout, state)
+            is CardReaderOnboardingViewModel.OnboardingViewState.WCStripeError ->
+                showWCStripeError(layout, state)
         }.exhaustive
     }
 
-    private fun showLoadingState(
+    private fun showLoadingState(view: View) {
+        val binding = FragmentCardReaderOnboardingLoadingBinding.bind(view)
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt:78]: It looks like that LoadingState has properties, but they are not used?

```
        object LoadingState : OnboardingViewState(R.layout.fragment_card_reader_onboarding_loading) {
            val headerLabel: UiString =
                UiString.UiStringRes(R.string.card_reader_onboarding_loading)
            val hintLabel: UiString =
                UiString.UiStringRes(R.string.please_wait)
            @DrawableRes
            val illustration: Int = R.drawable.img_payment_onboarding_loading
        }

 ```

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt [hunk 2]
```diff
@@ -61,28 +63,55 @@ class CardReaderOnboardingFragment : BaseFragment(R.layout.fragment_card_reader_
         binding.container.addView(layout)
```
Comment (kidinov) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt:63]: Just an idea - you could use ViewFlipper with included layouts you use for this flow. Then here you don't need to remove/add view, but you could do viewFlipper.setView(state.layoutRes). Also, things like that won't be needed anymore `FragmentCardReaderOnboardingLoadingBinding.bind(view)`. And transition animation (if needed) also comes from the box then

Definitely, not part of this PR though!

cc @malinajirka

**Review by kidinov — APPROVED** [review]
Looks good! Just a general comment which definitely should not be address in this PR and another one that, also not really connected to PR but if you want you can address it or not

**Comment by malinajirka** [comment]
Thanks for working on this @nbradbury!

> Note that I removed the button from the "pending requirements" state because that was the only state with a button, and it simply dismissed the fragment.

I believe we need to have this button there. When the account has "pending requirements" the user should be able to connect to a reader and accept in-person payments -> when the user taps on "dismiss" we should redirect them to "CardReaderDetail" screen. For all the other errors, in-person payments will not work so we should not let them use the feature. When I'm thinking about it, I'd consider replacing "Dismiss" with "Skip for now", wdyt?

P.S. If you agree with my reasoning, feel free to add the button in a separate PR.

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt (hunk 2 above)
Comment (malinajirka) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt:63]: Thanks for thinking even about these small but nice improvements! ;) However, I'm not sure if ViewFlipper is intended to be used for this type of use case. My understanding is that ViewFlipper is intended to be used to cycle through a set of views - either automatically or manually using `showNext/showPrevious` methods. Tbh I think `setView` or a similar method doesn't even exist. 
I think ViewSwitcher might be better for usecases where you want to change the currently displayed layout, however, it supports only two childs so it's not good for this use case either.
What do you think @kidinov ?

Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt:63]: The use of ViewFlipper is a very old trick so it works and battle-tested. The method which usually used is `setDisplayedChild` which accepts ~~layout id~~ position in the viewFlipper. 

There is [an article](https://medium.com/mobile-app-development-publication/viewflipper-for-state-transitioning-9baea6bcbd5) with a similar idea, but we actually don't need to extend anything, just extension function should work

**Comment by nbradbury** [comment]
> I believe we need to have this button there. When the account has "pending requirements" the user should be able to connect to a reader...

I had noticed the designs had buttons for a few of these screens, but none of that logic had been implemented so I assumed we were skipping that for now. My preference would be to add buttons in a separate PR, especially since that would probably require us to change the layout to a scrolling view (for landscape, at least).

File: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt (hunk 2 above)
Comment (nbradbury) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/ui/prefs/cardreader/onboarding/CardReaderOnboardingFragment.kt:63]: I'm also not sure about the idea of using a `ViewFlipper` here. If we were cycling between views that might work, but we're not really doing that here. I think we're fine with the current approach.

**Comment by malinajirka** [comment]
> I had noticed the designs had buttons for a few of these screens, but none of that logic had been implemented so I assumed we were skipping that for now. 

@nbradbury We are skipping all the other buttons (Call-to-actions) - but this particular button is not CTA. We should not prevent users from collecting payments in this scenario. (p91TBi-5uD-p2)

> My preference would be to add buttons in a separate PR, 

SGTM ;)

**Comment by nbradbury** [comment]
> this particular button is not CTA. We should not prevent users from collecting payments in this scenario.

Ah, got it. Should I open a separate issue for this?

**Comment by malinajirka** [comment]
@nbradbury I've added a task to the epic https://github.com/woocommerce/woocommerce-android/issues/4413.

**Pull request merged**